internal/crypto/cert.go

```diff
@@ -59,3 +65,48 @@ func ParseCertificates(data []byte, contentType string) (certs []*x509.Certifica
 	}
 	return parsePEM(data)
 }
+
+// CompleteCertificateChain returns the complete certificate chain by appending the parent certificate from the certificate bundle to the original certificate chain.
+// If the certificate bundle path is not set in the pluginConfig, it returns the original certificate chain.
+// If an error occurs while reading the certificate bundle or parsing its PEM, it returns a nil slice and a non-nil error indicating the failure reason.
+func CompleteCertificateChain(pluginConfig map[string]string, originalCerts []*x509.Certificate) ([]*x509.Certificate, error) {
+	certBundlePath, ok := pluginConfig[certBundleKey]
+	// if certbundle is not set, return the original certs
+	if !ok {
+		return originalCerts, nil
+	}
+	certBundlePath = strings.Trim(certBundlePath, " ")
+
+	certBundleBytes, err := os.ReadFile(certBundlePath)
+	if err != nil {
+		return nil, err
```

Comment:
We should provide a more detailed error message here, something like:
```suggestion
		return nil, fmt.Errorf("cannot open user provided cert bundle file %q with err: %w", certBundlePath, err)
```

Reply:
Updated.